.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,61 @@
+name: golangci-lint
+on:
+  #push:
+  #  branches:
+  #    - main
+  #    - 8.*
+  #    - 7.17
+  pull_request:
+permissions:
+  contents: read
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  pull-requests: read
+jobs:
+  golangci:
+    strategy:
+      matrix:
+        include:
+          - GOOS: windows
+          - GOOS: linux
+          - GOOS: darwin
```

Comment:
@rdner, is there any specific platform dependency while running golang-lint on windows and darwin? If so, I see down below the `runs-on: ubuntu-latest`, is that expected while running on windows/darwin too? I wonder if this matrix could be reduced to one OS (linux) if the outcome is the same for the other two platforms

Reply:
@v1v We have to run the linter against every platform (GOOS) we have any Go code for, otherwise any platform-specific code is not inspected and issues are not reported.

It does not matter on which OS it's running physically (`ubuntu-latest`), it's just an environment variable that tells the Go compiler which files to compile.

For Darwin we have these:

```
./metricbeat/internal/metrics/memory/memory_darwin.go
./metricbeat/internal/metrics/cpu/metrics_darwin.go
./auditbeat/module/file_integrity/fileorigin_darwin.go
./libbeat/statestore/backend/memlog/util_darwin.go
./libbeat/metric/system/process/process_darwin.go
./x-pack/osquerybeat/ext/osquery-extension/main_darwin.go
```

For Windows we have these:

```
./metricbeat/module/windows/service/zservice_windows.go
./metricbeat/module/windows/service/defs_service_windows.go
./metricbeat/internal/metrics/memory/memory_windows.go
./metricbeat/internal/metrics/cpu/metrics_windows.go
./metricbeat/internal/sysinit/system_windows.go
./metricbeat/helper/privileges_windows.go
./metricbeat/helper/dialer/dialer_windows.go
./metricbeat/helper/windows/pdh/zpdh_windows.go
./metricbeat/helper/windows/pdh/pdh_windows.go
./metricbeat/helper/windows/pdh/pdh_query_windows.go
./metricbeat/helper/windows/pdh/defs_pdh_windows.go
./filebeat/input/file/identifier_test_windows.go
./filebeat/input/file/identifier_inode_deviceid_windows.go
./filebeat/input/default-inputs/inputs_windows.go
./filebeat/input/filestream/fswatch_test_non_windows.go
./filebeat/input/filestream/identifier_inode_deviceid_windows.go
./filebeat/input/filestream/filestream_test_non_windows.go
./filebeat/input/filestream/identifier_test_non_windows.go
./auditbeat/module/file_integrity/fileinfo_windows.go
./auditbeat/module/file_integrity/security_windows.go
./auditbeat/module/file_integrity/zsecurity_windows.go
./winlogbeat/checkpoint/file_windows.go
./winlogbeat/sys/winevent/sid_windows.go
./winlogbeat/sys/wineventlog/syscall_windows.go
./winlogbeat/sys/wineventlog/zsyscall_windows.go
./winlogbeat/sys/wineventlog/wineventlog_windows.go
./winlogbeat/eventlog/errors_windows.go
./packetbeat/npcap/npcap_windows.go
./packetbeat/procs/syscall_windows.go
./packetbeat/procs/zsyscall_windows.go
./packetbeat/procs/procs_windows.go
./libbeat/logp/eventlog_windows.go
./libbeat/statestore/backend/memlog/util_windows.go
./libbeat/processors/decode_xml_wineventlog/decoder_windows.go
./libbeat/common/file/helper_windows.go
./libbeat/common/file/fileinfo_windows.go
./libbeat/common/file/stderr_windows.go
./libbeat/common/file/file_windows.go
./libbeat/api/npipe/listener_windows.go
./libbeat/api/make_listener_windows.go
./libbeat/service/service_windows.go
./libbeat/metric/system/numcpu/cpu_windows.go
./libbeat/metric/system/diskio/diskstat_windows.go
./libbeat/metric/system/process/process_windows.go
./x-pack/osquerybeat/ext/osquery-extension/main_windows.go
./x-pack/osquerybeat/internal/install/install_windows.go
./x-pack/osquerybeat/internal/osqd/osqueryd_windows.go
./x-pack/auditbeat/module/system/package/package_windows.go
./x-pack/winlogbeat/module/testing_windows.go
./x-pack/packetbeat/npcap/npcap_windows.go
./x-pack/libbeat/common/proc/job_windows.go
```

Reply:
Thanks for the clarification! It makes totally sense now :)